### PR TITLE
Mixin -> HOC + ES6 classes

### PR DESCRIPTION
**Pull request by retyui**
* Replaced mixin to hocs
* Removed react-create-call, Instead, ES classes are used.
* Added support new React Refs API

_the only difference from the version with myxins that the colors of the fill are different_

### CODE REVIEW

**Comment by retyui** [comment]
https://github.com/codesuki/react-d3-components/issues/120

**Comment by retyui** [comment]
https://github.com/codesuki/react-d3-components/issues/133

**Comment by retyui** [comment]
TODO:
- Add typescript type decloaration https://github.com/codesuki/react-d3-components/issues/103 |  [first commints](https://github.com/retyui/react-d3-components/tree/ts) 
- Update d3 to latest version https://github.com/codesuki/react-d3-components/issues/155
- Improve performance https://github.com/codesuki/react-d3-components/issues/23 | https://github.com/codesuki/react-d3-components/issues/105

**Comment by codesuki** [comment]
Sorry for the long delay. I made a branch for you:
https://github.com/codesuki/react-d3-components/tree/development
Good job so far!

**Comment by retyui** [comment]
@codesuki 
Now I'm finishing the tests.
And after that you can check PR

**Comment by codesuki** [comment]
Let me know when I should review!

File: .babelrc [hunk 1]
```diff
@@ -4,7 +4,8 @@
         [
             "@babel/preset-stage-0",
             {
-                "decoratorsLegacy": true
+                "decoratorsLegacy": true,
+                "pipelineProposal": "minimal"
```
Comment (jmpabon) [inline .babelrc]: Why do you need to add pipelineProposal ? I wouldn't go for stage-0 implementations unless we really really need to for some reason.

File: .eslintrc [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "rules": {
-        "quotes": [2, "single"],
+        "quotes": [1, "single"],
```
Comment (jmpabon) [inline .eslintrc]: This is not really part of your story. If you want to change the ESLint rules I would do a separate PR.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+sudo: false
```
Comment (jmpabon) [inline .travis.yml]: Why adding a travis yml ?

File: .npmignore [hunk 4]
```diff
@@ -1,9 +0,0 @@
-example
```
Comment (jmpabon) [inline .npmignore]: Why has this been removed?
